app-react/src/main/kotlin/com/ustadmobile/view/CourseDiscussionBlockEditScreen.kt

```diff
@@ -0,0 +1,194 @@
+    var onClickAddItem: () -> Unit
+    var onCourseBlockChange: (CourseBlock?) -> Unit
+}
+
+val CourseDiscussionBlockEditComponent2 = FC<CourseDiscussionBlockEditProps> { props ->
+
+    val strings = useStringsXml()
+
+    Container {
+        maxWidth = "lg"
+
+        Stack {
+            direction = responsive(mui.material.StackDirection.column)
+            spacing = responsive(10.px)
+
+            UstadTextEditField {
+                value = props.uiState.courseDiscussion?.courseDiscussionTitle ?: ""
+                label = strings[MessageID.title]
+                error = props.uiState.courseDiscussionTitleError
+                enabled = props.uiState.fieldsEnabled
+                onChange = {
+                    //TODO
+                }
+            }
+
+            UstadTextEditField {
+                value = props.uiState.courseDiscussion?.courseDiscussionDesc ?: ""
+                label = strings[MessageID.description]
+                error = props.uiState.courseDiscussionDescError
+                enabled = props.uiState.fieldsEnabled
+                onChange = {
+                    //TODO
+                }
+            }
+
+            UstadCourseBlockEdit {
+                uiState = props.uiState.courseBlockEditUiState
+                onCourseBlockChange = props.onCourseBlockChange
+            }
+
+
+            /*
+                onChange = {
+                    props.onSiteChanged(
+                        props.uiState.site?.shallowCopy {
+                            siteName = it
+                        })
+                }
+
+                 */
+        }
+
+
+
+        Typography {
+            variant = TypographyVariant.h6
+            +strings[MessageID.posts
+            ]
+        }
+
+        List {
+            ListItem {
+                disablePadding = true
+
+                ListItemButton {
+                    onClick = {
+                        props.onClickAddItem()
+                    }
+
+                    ListItemIcon {
+                        Add {}
+                    }
+
+                    ListItemText {
+                        +(strings[MessageID.posts])
+                    }
+                }
+            }
+
+            props.uiState.posts.forEach { item ->
+                ListItem {
+                    disablePadding = true
+                    secondaryAction = IconButton.create {
+                        onClick = {
+                            props.onDeleteClick(item)
+                        }
+                        Delete {}
+                    }
+
+                    ListItemButton {
+                        ListItemIcon {
+                            UstadBlankIcon { }
+                        }
+
+                        onClick = {
+                            props.onClickPost(item)
+                        }
+
+
+                        val dateFormatted = useMemo(dependencies = arrayOf(item.postLatestMessageTimestamp)) {
```

Comment:
This is the correct use of dependencies for the useMemo function. Better to use the function called "useFormattedDate" (which I added). This will use the locale set by the user (which works correctly when the user chooses a different locale to the browser default e.g. browser set to English, user selects Tajik). 

useFormattedDate requires the timezone. You can use the system default timezone via the new kotlinx datetime 

```
import kotlinx.datetime.TimeZone

TimeZone.currentSystemDefault().id
```